build_scripts/changed_functions.py

```diff
@@ -14,42 +14,43 @@ def parse_args():
         nargs=1,
         help="The folders to monitor for deployment to cognite functions",
     )
-    parser.add_argument("deploy_all", type=str, nargs="?", help="The folders to deploy to all ", default=None)
+    parser.add_argument("deploy_all", type=str, nargs="?", help="The folders to deploy to all", default=None)
     return parser.parse_args()
 
 
 def main():
     args = parse_args()
-    print(f"Got function folders: {args.folders!r}")
-    print(f"And deploy all folders: {args.deploy_all!r}")
-    function_folders = {f.strip() for f in args.folders[0].split(",")}
-    deploy_all_folders = set()
+    function_folders = sorted({f.strip() for f in args.folders[0].split(",")})
+    print(f"Input: Function folders to consider (re)deploying: {function_folders}")
+
+    deploy_all_folder = None
     if args.deploy_all:
-        deploy_all_folders = {f.strip() for f in args.deploy_all.split(",")}
-
-    changed_folders = subprocess.check_output("git diff --name-only HEAD^ HEAD".split(), text=True).split()
-    changed_folders = {
-        "/".join(parts if clean.name != "schedules" else parts.parts)
-        for f in changed_folders
-        if (clean := Path(f).parent) and clean.exists() and (parts := clean.parts)
-    }
-    print(f"Detected changed folders: {sorted(changed_folders)}")
-    deploy = []
-    for changed_folder in changed_folders:
-        if any(changed_folder.startswith(deploy_all) for deploy_all in deploy_all_folders):
-            deploy = list(function_folders)
-            break
-        if changed_folder in function_folders:
-            deploy.append(changed_folder)
-
-    if not deploy:
-        deploy = ["skipDeploy"]
-
-    json_format = json.dumps({"folders": deploy})
+        deploy_all_folder = args.deploy_all.strip()
+        print(f"Input: Common folder (may force deploy all): {deploy_all_folder!r}")
+
+    # Compare against previous commit under the assumption of squash-only merges:
+    diff = subprocess.check_output("git diff --name-only HEAD^ HEAD".split(), text=True).split()
+    changed_files = set(map(Path, diff))
```

Comment:
It always take me a few seconds extra to read `map`, more pythonic to use a comprehension?
```suggestion
    changed_files = {Path(path) for path in diff}
```

Reply:
👀